server/src/main/java/org/eclipse/hono/server/MessageForwardingEndpoint.java

```diff
@@ -38,13 +40,24 @@
  */
 public abstract class MessageForwardingEndpoint extends BaseEndpoint {
 
+    protected CounterService              counterService;
```

Comment:
In particular it seems to be an *optional* cross cutting concern. Alternatively we could use an *empty* default implementation of the used gauges and counters which simply do nothing but ignoring the values. WDYT?